src/lib/lip-sync-analyzer.ts

```
@@ -19,135 +19,194 @@ export interface LipSyncData {
-    } catch (error) {
-      console.error('Error analyzing lip-sync:', error);
-      throw error;
-    }
+        clearTimeout(timeoutId);
+        resolve(lipSyncData);
+      } catch (error) {
+        clearTimeout(timeoutId);
+        console.error('Error analyzing lip-sync:', error);
+        reject(error);
+      }
+    });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)